src/components/ImportForm/utils/validation-utils.ts

```diff
@@ -0,0 +1,73 @@
+import * as yup from 'yup';
+import { createApplication } from '../../../utils/create-utils';
+
+export const gitUrlRegex =
+  /^((((ssh|git|https?:?):\/\/:?)(([^\s@]+@|[^@]:?)[-\w.]+(:\d\d+:?)?(\/[-\w.~/?[\]!$&'()*+,;=:@%]*:?)?:?))|([^\s@]+@[-\w.]+:[-\w.~/?[\]!$&'()*+,;=:@%]*?:?))$/;
+
+// generic regex to validate container image /^[^/]+\.[^/.]+\/([a-z0-9-_]+\/)?[^/.]+(:.+)?$/
+export const containerImageRegex = /^(https:\/\/)?quay.io\/([a-z0-9-_]+\/)?[^/.]+(:.+)?$/;
+
+export const componentNameRegex = /^[a-z0-9]([-a-z0-9]*[a-z0-9])?$/;
+
+const combineRegExps = (...regexps: RegExp[]) => {
+  const regexStringsWithoutFlags = regexps.map((regex) => regex.source);
+  return new RegExp(regexStringsWithoutFlags.join('|'));
+};
+
+export const applicationValidationSchema = yup.object({
+  application: yup
+    .string()
+    .required('Required')
+    .test('isUnique', 'Application name already exists.', async (value, context) => {
+      try {
+        await createApplication(value, context.parent.namespace, true);
+        return true;
```

Comment:
Can this be debounced? Right now it's running on every keystroke

Reply:
Okay, using `validateOnChange` in the step.
But not sure if it's working, I can see multiple network calls on every key press.